test/FieldArray.test.tsx

```diff
@@ -37,6 +37,26 @@ describe('<FieldArray />', () => {
     );
   });
 
+  it('renders component and passes extra props down', () => {
+    const TestComponent = (props: any) => {
+      expect(props.userProps === 'test');
+      return null;
+    };
+
+    ReactDOM.render(
+      <TestForm
+        component={() => (
+          <FieldArray
```

Comment:
should be `<FieldArray<ExtraProps> />` which uses all the ExtraProps declarations I commented above.

Reply:
I was able to write the types above pretty easily, but `FieldArray`, and `Field` do not accept generic parameters right now so I'm going to see if I can figure that part out tonight maybe

Reply:
I have a PR where I implemented `Field` strong types and am in the process of updating it for v2. I didn't touch FieldArray though. Another individual also has a Field typing PR

#1336 
#1786